### DISCUSSION TITLE
When will pydantic 2.0 be supported

### DISCUSSION THREAD

**Discussion by lpdswing** [BeanieODM/beanie#616] [Question]
When will pydantic 2 be supported

**Comment by roman-right** [comment]
Hi! Next week I'll release the beta version. One week after I'll publish the release version.

**Reply by pythonhubdev** [reply]
Hi @roman-right I would like to help with this in anyways I could I have been trying to migrate beanie to pydantic v2 but I only got 30% of test coverage and also most of the tests failed don't know the exact reason but can you share which branch you are using to commit the latest code so that I could learn what I did wrong.

**Reply by roman-right** [reply]
Hi @pythonhubdev,

Yes, sure. Here is the branch: https://github.com/roman-right/beanie/tree/bump/pydantic.
Please be aware that it has some ugly fixes.

My plan is to do the following in order:

- Make it pass all the tests on Python 3.10.
- Ensure it is compatible with all supported Python versions.
- Check the number and size of fixes and decide whether to support both Pydantic v1 and v2 in the same version or publish different versions for different Pydantic versions. Probably supporting both interfaces together will be too complicated.
- Refactor the code.
- Release.

Between each step, I will publish alpha/beta versions to PyPI.

**Reply by pythonhubdev** [reply]
Thanks a lot

**Comment by keysking** [comment]
My VSCode say: Argument missing for parameter "self".
Is this because of the pydantic version? And can the beanie support pydantic2 solve this problem?
![image](https://github.com/roman-right/beanie/assets/16982187/0b977c08-1623-46b8-b016-b08aba272e4a)

and this:
![image](https://github.com/roman-right/beanie/assets/16982187/672f8a1e-3e77-487a-b89d-2f15f679fb0d)

**Reply by keysking** [reply]
Sorry, I made a mistake. My pydantic version is 1.10.11
but... why vscode say that?

python version is 3.10.12
beanie version is 1.20.0

**Reply by roman-right** [reply]
Regarding Indexed - please use `# type: ignore` or something else Pylance-specific. It cannot be fixed soon. Alternatively, you can set indexes in the `Settings` inner class.

Regarding self - it looks weird. I'll check this with the Pylance documentation.

I stopped supporting Pylance a year ago because, after each Pylance update, more and more annotations had to be fixed, usually with workarounds. I plan to fix Pylance issues once, but I don't know when yet.

**Reply by keysking** [reply]
Thank you for your reply. # type: ignore is ok. :smile:
